orc8r/cloud/go/services/service_registry/servicers/docker_servicer.go

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package servicers
+
+import (
+	"fmt"
+	"strings"
+
+	"magma/orc8r/cloud/go/services/service_registry"
+	"magma/orc8r/lib/go/protos"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"golang.org/x/net/context"
+)
+
+const (
+	// Use custom label to select only orc8r applications
+	orc8rServicePartOfLabel         = "part-of=orc8r"
+	dockerComposeServiceLabelPrefix = "com.docker.compose.service"
+)
+
+type DockerClient interface {
+	ContainerList(ctx context.Context, options types.ContainerListOptions) ([]types.Container, error)
+}
+
+type DockerServiceRegistryServicer struct {
+	dockerClient DockerClient
+}
+
+// NewDockerServiceRegistryServicer creates a new service registry servicer
+// that is backed by Docker.
+func NewDockerServiceRegistryServicer(dockerCli DockerClient) *DockerServiceRegistryServicer {
+	return &DockerServiceRegistryServicer{
+		dockerClient: dockerCli,
+	}
+}
+
+// ListAllServices returns the service name of all services in the registry.
+func (s *DockerServiceRegistryServicer) ListAllServices(ctx context.Context, req *protos.Void) (*protos.ListAllServicesResponse, error) {
+	labelFilter := filters.NewArgs()
+	labelFilter.Add("label", orc8rServicePartOfLabel)
+	listAllFilter := types.ContainerListOptions{
+		All:     true,
+		Filters: labelFilter,
+	}
+	allContainers, err := s.dockerClient.ContainerList(context.Background(), listAllFilter)
+	if err != nil {
+		return &protos.ListAllServicesResponse{}, err
+	}
+	services := []string{}
+	for _, container := range allContainers {
+		if len(container.Names) == 0 {
+			continue
+		}
+		serviceName := container.Names[0]
+		serviceName = strings.TrimPrefix(serviceName, "/")
+		services = append(services, serviceName)
+	}
+	return &protos.ListAllServicesResponse{
+		Services: services,
+	}, nil
+}
+
+// FindServices returns all services in that have the provided label.
+func (s *DockerServiceRegistryServicer) FindServices(ctx context.Context, req *protos.FindServicesRequest) (*protos.FindServicesResponse, error) {
+	if req == nil {
```

Comment:
Not sure if this is an implementation of GRPC service. But if so, is it possible req is ever nil, and we may not need this? Will not the GRPC client always create a default req?

If not GRPC implementation, then disregard the comment